### PR TITLE
Closed valves and pumps and vents now block ventcrawling and other tweaks

### PR DESCRIPTION
**Pull request by mystery3525**
## About The Pull Request

- Ventcrawling now makes a little bit more sense:
  - Vents and scrubbers: Possible as long as the exit/entrance isn't welded and they aren't broken
  - Passive Vents: Always possible
  - Manual, Digital, and Pressure Valves: Possible as long as they are opened
  - Regular and Volume pumps: Possible as long as they are on
  - Every other atmosmachine: Possible as long as they are on AND powered, OR they are depowered (via APC).
- There is now a 1% chance that each time a ventcrawling sound plays (3 seconds of movement), a visible message with `You hear something crawling through the ducts...`

## Why It's Good For The Game

To me it feels like if a valve is shut, nothing should be able to go through it, right? I extended this to more atmospheric devices, so now we can't bypass logic here. And this will finally make those KiloStation cutoff valves useful (maybe?)

Robust players already knows the vent crawling sound means something is in the vents, this opens up the door for more players being telegraphed information (if they even read the bottom right window these days) and means that vent-crawing creatures need to get a move on if they are trying to be stealthy to avoid detection. The 1% chance amounts to about one message every five minutes of constant movement in the vents, so I believe it's fair enough for the creature in the vents.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/30960302/a04cb099-839e-47f3-8a60-ffdab97cb0b5

https://github.com/BeeStation/BeeStation-Hornet/assets/30960302/5eff2582-6700-4b67-940c-73c6b9ce3f8a

</details>

## Changelog
:cl:
tweak: Vent crawling now has a 1% chance every 3 seconds of constant movement of giving notice to everyone nearby that can hear it, meaning once per 5 minutes of vent crawling does it play a sound to hearing mobs.
balance: Closed/broken/offline atmospherics machinery now prevents creatures from vent-crawling through them, though vents/scrubbers remain unchanged and require being welded still.
/:cl:

### CODE REVIEW

**Comment by LodedDiper69** [comment]
Has it been taken into consideration how this affects antags that rely on ventcrawling?

**Comment by Tsar-Salat** [comment]
Man, I really hope this stays a neat little feature and isn't metagamed :trollface:

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 1]
```diff
@@ -317,6 +317,8 @@
 				if(world.time - user.last_played_vent > VENT_SOUND_DELAY)
 					user.last_played_vent = world.time
 					playsound(src, 'sound/machines/ventcrawl.ogg', 50, 1, -3)
+					if(prob(1))
+						visible_message("<span class='warning'>You hear something crawling through the ducts...</span>", ignored_mobs = list(user))
```
Comment (PowerfulBacon) [inline code/modules/atmospherics/machinery/atmosmachinery.dm]: This shouldn't show to deaf people

**Comment by PowerfulBacon** [comment]
I'm not sure if I agree with the sound effect going to chat, I think its one of the things that is better left uncertain with a sound that can be easilly confused

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
Any of these which require power to function should be possible for ventcrawlers to bypass when they are unpowered, especially vents and scrubbers.

They should be possible to bypass even if closed when depowered

**Comment by Rukofamicom** [comment]
While I agree that it makes sense, I do not agree with the balance implications of the vent/scrubber portion of this without giving antagonists that heavily rely on this a way around it. 

This is a massive blow to antagonists that heavily rely on ventcrawling because of the ease and speed with which silicons (AI especially) can now entirely shut down ventcrawling antagonists. Xenos, thanks to their sheer power will probably be fine, but this is a potentially fatal blow to spider balance for instance.

**Review by Rukofamicom — COMMENTED** [review]
With the requested change implemented, I'm cautiously optimistic about this.

I think this will need a testmerge period to observe exactly how things play out, and may have additional requests afterwards.

**Comment by mystery3525** [comment]
Also updated the CL to be more intuitive to read

**Review by PowerfulBacon — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
This is fine without a testmerge I think

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 2]
```diff
@@ -329,8 +331,17 @@
...
-
+/// Whether ventcrawling creatures can move in or out of this machine.
 /obj/machinery/atmospherics/proc/can_crawl_through()
+	// If it's broken, nothing can pass through
+	if(machine_stat & BROKEN)
+		return FALSE
+
+	// If it is powered on, but off, nothing can pass through
+	if(powered() && !on)
+		return FALSE
```
Comment (PowerfulBacon) [inline code/modules/atmospherics/machinery/atmosmachinery.dm:342]: This makes no sense, why would someone not be able to crawl through a scrubber or vent that is turned off?

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/machinery/atmosmachinery.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/modules/atmospherics/machinery/atmosmachinery.dm:342]: If people want to block vent crawlers, they should have to weld the vents

Comment (Rukofamicom) [reply code/modules/atmospherics/machinery/atmosmachinery.dm:342]: If it is powered and turned off it is presumably closed to the point of being air tight.

**Review by Rukofamicom — APPROVED** [review]

File: code/modules/atmospherics/machinery/atmosmachinery.dm (hunk 2 above)
Comment (mystery3525) [reply code/modules/atmospherics/machinery/atmosmachinery.dm:342]: the vent has motors that (while being powered) will force air flaps to close, meaning nothing can go through them

**Comment by PowerfulBacon** [comment]
The changes to things like vents and scrubbers not allowing vent crawling while turned off don't make a lot of sense and make the AI able to 'weld' the entire station almost instantly. This is far too powerful of a blocker for things that depend on ventcrawling and doesn't have the same cost of requiring someone to physically access the vent to block it.

I would be fine for this on valves & (Volume/pipe) pumps only.

**Comment by Howluinb** [comment]
Apart from AI shutting off APCs, i remember breaking apcs causing vents/scrubbers to turn off... which would be terrible for *all* ventcrawling antags

**Comment by Rukofamicom** [comment]
The description needs to be updated.

Depowered vents ands scrubbers are accessible to ventcrawlers now.

**Comment by PowerfulBacon** [comment]
Looking at the code, vent scrubbers are still blocked if depowered as they call ..()

**Comment by PowerfulBacon** [comment]
Oh, depowered vents are accessible. I want it so that they are still vent crawlable if off even if off has power.

**Comment by mystery3525** [comment]
I have made a few changes that may be better
- Vents and scrubbers: Possible as long as the exit/entrance isn't welded and isn't broken
- Passive Vents: Always possible
- Manual, Digital, and Pressure Valves: Possible as long as they are opened
- Regular and Volume pumps: Possible as long as they are on
- Every other atmosmachine: Possible as long as they are on AND powered, OR they are depowered (via APC).

**Review by PowerfulBacon — APPROVED** [review]

File: code/modules/atmospherics/machinery/components/binary_devices/valve.dm [hunk 3]
```diff
@@ -40,6 +40,10 @@ It's like a regular ol' straight pipe, but you can turn it on and off.
 		flick("[valve_type]valve_[on][!on]-[set_overlay_offset(piping_layer)]", src)
 	icon_state = "[valve_type]valve_[on ? "on" : "off"]-[set_overlay_offset(piping_layer)]"
 
+/obj/machinery/atmospherics/components/binary/valve/can_crawl_through()
+	. = ..()
+	return . && on // valves should block whatever is trying to go through them, regardless of power
```
Comment (PowerfulBacon) [inline code/modules/atmospherics/machinery/components/binary_devices/valve.dm]: This one will respect being powered despite being a valve. Once closed, it is closed regardless of power

**Pull request merged**